src/main/kotlin/uk/gov/justice/hmpps/offendersearch/services/MatchService.kt

```diff
@@ -29,25 +30,44 @@ class MatchService(
 
   fun match(matchRequest: MatchRequest): OffenderMatches {
     matchBy(matchRequest) { nomsNumber(it) } onMatch { return it.matches }
+    matchBy(matchRequest) { croNumber(it) } onMatch { return it.matches }
     matchBy(matchRequest) { pncNumber(it) } onMatch { return it.matches }
 
     // TODO remove this fallback once the full matching algorithm has been completed
-    matchBy(matchRequest) { fallback(it)} onMatch { return it.matches }
+    matchBy(matchRequest) { fallback(it) } onMatch { return it.matches }
 
     return OffenderMatches()
   }
 
   private fun nomsNumber(matchRequest: MatchRequest): BoolQueryBuilder? {
-    return matchRequest.nomsNumber.takeIf { !it.isNullOrBlank() } ?.let {
+    return matchRequest.nomsNumber.takeIf { !it.isNullOrBlank() }?.let {
+      // NOMS number is a special case since a human has already matched so trust that judgement
       return QueryBuilders.boolQuery()
           .mustWhenPresent("otherIds.nomsNumber", it)
     }
   }
 
   private fun pncNumber(matchRequest: MatchRequest): BoolQueryBuilder? {
-    return matchRequest.pncNumber.takeIf { !it.isNullOrBlank() }?.let {
-      return QueryBuilders.boolQuery()
-          .mustMultiMatchKeyword(it.canonicalPNCNumber(), "otherIds.pncNumberLongYear", "otherIds.pncNumberShortYear")
+    with(matchRequest) {
+      return pncNumber.takeIf { !it.isNullOrBlank() }?.let {
+        return QueryBuilders.boolQuery()
+            .mustMultiMatchKeyword(it.canonicalPNCNumber(), "otherIds.pncNumberLongYear", "otherIds.pncNumberShortYear")
+            .must(QueryBuilders.boolQuery()
+                .shouldWhenPresent("surname", surname)
```

Comment:
You said that should was coming but I don't understand the difference between must and should in a query context... Please can you elaborate just for my understanding :)

Reply:
Yep; in the Elastic Search language "must" = "and" while "should" = "or" therefore this statement says:

Match = "PNC Number" AND ("SURNAME" OR "DATE OF BIRTH")

In fact the JSON it creates looks something not too far from this.

By the way multi-match is similar; but means this "phrase" must match in one of these fields. So given the following
PNC = "2018/12345x"
SURNAME = "Marke"
DOB = "1965-07-19"

The expression above would be
```

WHERE (pncNumberLongYear = 2018/12345x OR pncNumberShortYear = 2018/12345x) 
AND (surname = MARKE OR dateOfBirth =  1965-07-19)

```

Reply:
An interesting choice of semantics!  Thanks Andy.